### ISSUE TITLE
Any "git ls-files --stage" analogue?

### ISSUE THREAD

**Issue by iliakan** [isomorphic-git/isomorphic-git#1726]
In case of conflict, git stores multiple versions of file in index.

Here's an example:
```
➜  tmpgit git:(master) ✗ git ls-files --stage
100644 e8e448b33545a4ff26011d77ccc46c0b879f50be 1	index.html
100644 da9736cb76038951b7931f826336a23d21ffb03a 2	index.html
100644 f4952f928d1644e8bfcccc5c96cee07712d7dbaa 3	index.html
100644 ec0966979aa7da1907f1c205b87c173050d7a223 0	script.js
```

How can I get similar information from isomorphic-git?

**Comment by iliakan** [comment]
I tried to walk, like this, but I only see `index.html` once here, not 3 times:
```
await git.walk({
    fs,
    dir: '/',
    trees: [git.STAGE()],
    map: async function(filePath, [A]) {
       ...
    }
  });
```

In other words, the walker doesn't show all 3 entries of `index.html` in the index.

**Comment by iliakan** [comment]
After going deeper: the answer is NO, because `GitIndex` doesn't store them.

Proposed to fix in #1727.
